packages/data-context/src/sources/migration/codegen.ts

```diff
@@ -435,6 +435,13 @@ export function reduceConfig (cfg: LegacyCypressConfigJson, options: CreateConfi
           component: { ...acc.component, excludeSpecPattern: val },
         }
       case 'supportFile':
+        // If the supportFile is set, but is the same value as the default one; where
+        // we migrate it, we do not want to put the legacy value in the migrated config.
+        // It can be .ts or .js
+        if (val.includes('cypress/support/index.')) {
```

Comment:
@estrada9166 Is this also going to include people's support file being set to `tests/cypress/support/index.js`? Because we only want it to handle exact values matching our default value in this way, so it should start with `cypress/support/index.js|ts`.

Reply:
Maybe a regex match for:
```ts
/^cypress\/support\/index\.(ts|js)$/
```

Reply:
see my comment here: see my comment here: https://github.com/cypress-io/cypress/pull/21985#issuecomment-1142592792

Reply:
this PR needs to cover all the situations that are valid values that still resolve the same underlying file
- if i wrote `cypress/support` it would resolve `cypress/support/index.js`
- if i wrote `cypress/support/index` it would resolve `cypress/support/index.js`
- if i wrote `cypress/support/index.js` it would resolve `cypress/support/index.js`